lib/metadata/hfbMetadata.hpp

```diff
@@ -2,14 +2,11 @@
 #define HFB_METADATA
 
 #include "buffer.h"
+#include "dataset.hpp" // for dset_id_t
 #include "metadata.h"
 
 #include <sys/time.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #pragma pack()
```

Comment:
Why change the packing? Is there a good reason to not just use the standard C packing. And if you're going to change it from the default, I think it's at least good to be explicit. I think `#pragma pack()` with not argument sets 8 byte alignment.

If you're worried about wasted space in the struct, you can get rid of it much more transparently by reordering the entries.

Reply:
I had a problem compiling with the `extern "C"`. I've just tried it again and it compiles okay now, so I'll revert this change.

Reply:
I am now getting this error in the MacOS build:
```
../../../lib/metadata/HFBMetadata.h:93:18: error: 'get_dataset_id' has C-linkage specified, but returns user-defined type 'dset_id_t' (aka 'Hash') which is incompatible with C [-Werror,-Wreturn-type-c-linkage]
inline dset_id_t get_dataset_id(struct Buffer* buf, int ID) {
                 ^
1 error generated.
```

have you seen this before? I think this is why I changed the file to C++ in the first place.

Reply:
Yeah, I think this definitely needs to be C++. However, I don't know why you need to declare is as `extern "C"`. Do you call it from any C code?

My initial comment though, is why you change the struct packing and alignment by using the `#pragma pack()` line? That wasn't a change you made this round, but I don't see why you need to do it at all, and I'd suggest removing it entirely if there isn't a good reason.

Reply:
Ah okay, I will make it `.hpp` file again. The `extern "C"` and `#pragma pack()` just came from copying the `chimeMetadata.h` so I will remove them.

Reply:
I guess that turns into a question for @andrerenard. What's the point of changing the default struct packing in chimeMetadata? Are you trying to align things in memory more efficiently by using the (implicit) 8 byte alignment of members.

Reply:
The main reason for this was trying to force the compiler's hand in the packing scheme to avoid possibly different packings on different systems.  However in practice this might not isn't really an issue for our use cases, and in theory not forcing packing can be faster.

Reply:
Okay. That makes some sense, although I can't imagine trying to run kotekan on any platform which didn't have the standard System V x64 ABI, so I don't think it's really helping. I guess it could be removed in chimeMetadata, but then we'd be unable to load any historic raw dumps we had lying around.

I think at least for the HFB case we should definitely remove this one though.